fix/workorder-detail-fix.lua

```diff
@@ -0,0 +1,199 @@
+        if type(order_flags) == 'number' then
+            job_item[flags] = order_flags
+        else
+            job_item[flags].whole = order_flags.whole
+        end
+    end
+end
+
+-- correct job's job_items to match the order's
+local function enforce_order_details(job, order)
+    if not job.job_items then return end -- never happens (error here?)
+    local modified = false
+    for idx, job_item in ipairs(job.job_items) do
+        local order_item = order.items[idx]
+        if not order_item then break end -- never happens (error here?)
+        if job_item.item_type ~= order_item.item_type then
+            -- dfhack's isSuitableItem function will allow the orders we want,
+            -- but disallow insane combinations like meals made of shoes
+            local suitable = dfhack.job.isSuitableItem(
+                job_item, order_item.item_type, order_item.item_subtype )
+            if suitable then
+                correct_item_details(job_item, order_item)
+                modified = true
+            else --[[ error on unsuitable item?]] end
+        end
+    end
+    if modified then jobs_corrected = jobs_corrected + 1 end
+end
+
+-- check orders list to see if fix is needed
+-- yields id-order table for the orders with issues, or nil when no issues
+local PrepareMeal = df.job_type.PrepareMeal
+local SewImage = df.job_type.SewImage
+local NONE = df.job_type.NONE
+local function get_bugged_orders()
+    local orders = {}
+    local num_bugged, improve_item_index, item = 0, 1, nil
+    for _, order in ipairs(df.global.world.manager_orders) do
+        if not order.items then goto nextorder end
+        local order_job_type = order.job_type
+        if not offending_jobs[order_job_type] then goto nextorder end
+        if #order.items == 0 then goto nextorder end -- doesn't happen
+
+        -- for PrepareMeal jobs, any one of the items could be an issue.
+        if order_job_type == PrepareMeal then
+            for _, _item in ipairs(order.items) do
+                if _item.item_type ~= NONE then goto fix end
+            end
+            goto nextorder
+        end
+
+        -- All other types are improve jobs; only the improved item is checked
+        -- Only SewImage has the item-to-improve at items[0]
+        improve_item_index = (order_job_type ~= SewImage) and 1 or 0
+        item = order.items[improve_item_index]
+        if not item then goto nextorder end -- error here?
+        if item.item_type == NONE then goto nextorder end
+        :: fix ::
+        num_bugged = num_bugged + 1
+        orders[order.id] = order
+        :: nextorder ::
+    end
+    if num_bugged == 0 then return nil end
+    return orders
+end
+
+-- correct newly dispatched work order jobs
+local disable
+local function on_dispatch_tick()
+    if df.global.cur_year_tick % 150 ~= 30 then
+        print(script_name.." desynced from dispatch tick")
+        repeatutil.cancel(schedule_key)
+        disable(true)
```

Comment:
got it re-syncing now. Surprisingly the fix works with it on. I wouldn't think `timestream` handles work orders well in the first place

Also, the already-gathering-items "fix" is gone for now because I got a couple crashes (!), either relating to the general refs or the `while` loop checking the vector. It's not crucial anyway